### PR TITLE
A script to compile with Coverity Scan.

### PR DESCRIPTION
**Pull request by coryan**
Coverity Scan is a static analysis tool.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/1775)

### CODE REVIEW

File: ci/coverity-scan.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+# Copyright 2018 Google LLC
```
Comment (mbrukman) [inline ci/coverity-scan.sh]: 2019?

File: ci/coverity-scan.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+if [[ -z "$(which cov-build)" ]]; then
+  echo "This script requires the coverity scan tool (cov-build) in PATH"
+  echo "Please download the tool, make sure your PATH includes the directory"
+  echo "that contains it and try again."
+  echo "More details on: https://scan.coverity.com/download"
```
Comment (mbrukman) [inline ci/coverity-scan.sh]: For more details, see:

File: ci/coverity-scan.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  exit 1
+fi
+
+# Coverity scan is pre-configured to run with gcc, so do the same here.
```
Comment (mbrukman) [inline ci/coverity-scan.sh]: It doesn't support Clang? :disappointed:

File: ci/coverity-scan.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+export CXX=g++
+export CC=gcc
+
+# Running with coverity-scan and ccache seems like a bad idea, disable ccache.
+# Also build in Debug mode because this is too slow in Release mode.
```
Comment (mbrukman) [inline ci/coverity-scan.sh]: ... because building in Release mode takes too long.

File: ci/coverity-scan.sh [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+cmake --build .coverity --target curl_project -- -j $(nproc)
+cmake --build .coverity --target crc32c_project -- -j $(nproc)
+cmake --build .coverity --target googletest_project -- -j $(nproc)
+
+# The proto-generated files contain too many errors, and they are not
+# actionable, so they are built without coverity-scan too.
+cmake --build .coverity --target skip-scanbuild-targets -- -j $(nproc)
+
+# Run coverity scan over our code.
+cov-build --dir cov-int cmake --build .coverity -- -j $(nproc)
```
Comment (mbrukman) [inline ci/coverity-scan.sh]: Is there a step to upload results to Coverity? Or is this automatic/implicit in `cov-build`?

Do you need to also annotate what "version" this is for Coverity? It keeps track of findings and trends by version.

File: ci/coverity-scan.sh (hunk 1 above)
Comment (mbrukman) [inline ci/coverity-scan.sh]: Do you typically add a blank line to separate `#!` line from the license or no?

File: ci/coverity-scan.sh [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+if [[ -z "$(which cov-build)" ]]; then
+  echo "This script requires the coverity scan tool (cov-build) in PATH"
```
Comment (devbww) [inline ci/coverity-scan.sh]: s/PATH/PATH./ perhaps.

File: ci/coverity-scan.sh (hunk 1 above)
Comment (coryan) [reply ci/coverity-scan.sh]: More often than not we put an empty comment line.

File: ci/coverity-scan.sh (hunk 6 above)
Comment (coryan) [reply ci/coverity-scan.sh]: Done.

**Pull request merged**